Libplanet.Explorer/GraphTypes/CurrencyInputType.cs

```diff
@@ -0,0 +1,83 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Numerics;
+using GraphQL.Types;
+using Libplanet.Assets;
+
+namespace Libplanet.Explorer.GraphTypes
+{
+    public class CurrencyInputType : InputObjectGraphType<Currency>
+    {
+        public CurrencyInputType()
+        {
+            Name = "CurrencyInput";
+            Field<NonNullGraphType<StringGraphType>>(
+                "ticker",
+                "The ticker symbol, e.g., USD."
+            );
+            Field<NonNullGraphType<ByteGraphType>>(
+                "decimalPlaces",
+                "The number of digits to treat as minor units (i.e., exponents)."
+            );
+            Field<ListGraphType<NonNullGraphType<AddressType>>>(
+                "minters",
+                "The addresses who can mint this currency.  If this is null anyone can " +
+                "mint the currency.  On the other hand, unlike null, an empty set means no one " +
+                "can mint the currency."
+            );
+            Field<BigIntGraphType>("maximumSupplyMajorUnit");
+            Field<BigIntGraphType>("maximumSupplyMinorUnit");
+            Field<NonNullGraphType<BooleanGraphType>>(
+                "totalSupplyTrackable",
+                "Whether the total supply of this currency is trackable."
+            );
+        }
+
+        public override object ParseDictionary(IDictionary<string, object?> value)
+        {
+            IImmutableSet<Address>? minters = null;
+            var rawMinters = value.TryGetValue("minters", out object? obj) && obj is object[] o
+                ? o
+                : null;
+            if (rawMinters is not null)
+            {
+                if (rawMinters.Any())
+                {
+                    minters = ImmutableHashSet<Address>.Empty;
+                    foreach (var rawMinter in rawMinters)
+                    {
+                        minters = minters.Add((Address)rawMinter);
+                    }
+                }
+            }
+
+            bool totalSupplyTrackable = (bool)value["totalSupplyTrackable"]!;
+            (BigInteger, BigInteger)? maximumSupply =
+                value.TryGetValue("maximumSupplyMajorUnit", out object? nullableMajorUnit) &&
+                nullableMajorUnit is BigInteger majorUnit &&
+                value.TryGetValue("maximumSupplyMinorUnit", out object? nullableMinorUnit) &&
+                nullableMinorUnit is BigInteger minorUnit
+                    ? (majorUnit, minorUnit)
+                    : null;
```

Comment:
Instead of silently falling back to non-capped currency when only one of majorunit and minorunit is given, wouldn't it be better if it throws an exception when only majorunit is given or minorunit is given, similar to `public Currency(IValue serialized)` constructor?

Reply:
I applied your suggestion while rebasing this branch. Please take review it once more.

https://github.com/planetarium/libplanet/compare/c73507309acdb3ba10a2754f374b1c797d605bc6..7ae3bc5559f8e6b976f6a7e294668978dbc34fbe#diff-a5ce41d0818730ae36f072fbdabf14d96e2f086c8b3578c314928a8400ad01f9R45-R64